### ISSUE TITLE
Missing a memory overrun check in mwrite-box/symbol

### ISSUE THREAD

**Issue by i-perminov** [cosmos72/hyperluminal-mem#10]
%mwrite-package (called from mwrite-box/symbol) may write a package reference without doing a buffer boundary check.
Test case:
```
(let ((real-buf-size 4))
           (hlmem:with-mem-words (buf real-buf-size)
             ;; Zero the buffer
             (dotimes (i real-buf-size) (setf (cffi:mem-aref buf :uint64 i) 0))
             ;; Should fail with "attempt to write N words ..., but only M words available ...
             ;; buf fails with a type error
             (print (multiple-value-list (ignore-errors (hlmem:mwrite buf 0 1 :foo))))
             ;; Should be zero, because last-index is 1 and mwrite should not write past last-index-1.
             (print (cffi:mem-aref buf :uint64 1))))
```

**Comment by cosmos72** [comment]
Thanks for spotting this!
There was a missing overrun check in writing the package name of a symbol i.e. the internal function `(hlmem::%mwrite-package)`
Fixed in bfd71c681f5d6711809dafa9a59fb36ebe29c525

**Comment by cosmos72** [comment]
For reference, there are also two low-level functions useful for debugging:
```
(hlmem:mread-word ptr i)            ;; equivalent to (cffi:mem-aref ptr :uint64 i) ;; or :uint32 on 32-bit systems
(hlmem:mwrite-word ptr i value)   ;; equivalent to (setf (cffi:mem-aref ptr :uint64 i) value) ;; or uint32 on 32-bit systems
```
